sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_deserialize.py

```diff
@@ -4,39 +4,40 @@
 # license information.
 # --------------------------------------------------------------------------
 import logging
-from typing import NoReturn, TYPE_CHECKING
-from xml.etree.ElementTree import Element
-
-from azure.core.pipeline.policies import ContentDecodePolicy
-from azure.core.exceptions import (
-    HttpResponseError,
-    DecodeError,
-    ResourceModifiedError,
-    ClientAuthenticationError,
-    ResourceNotFoundError,
-    ResourceExistsError
+from typing import (
+    Any, cast, Collection, Dict, List, Tuple,
+    TYPE_CHECKING
 )
 from ._models import (
-    FileProperties,
+    AnalyticsLogging,
+    DeletedPathProperties,
     DirectoryProperties,
+    FileProperties,
     LeaseProperties,
-    DeletedPathProperties,
-    StaticWebsite,
-    RetentionPolicy,
     Metrics,
-    AnalyticsLogging,
-    PathProperties
+    PathProperties,
+    RetentionPolicy,
+    StaticWebsite
 )
-from ._shared.models import StorageErrorCode
 from ._shared.response_handlers import deserialize_metadata
 
 if TYPE_CHECKING:
-    pass
+    from azure.core.rest import RestRequestsTransportResponse  # type: ignore [attr-defined]
```

Comment:
What is this one? Not sure if there's a better way around this but seems a little silly that we have to import something for type-checking just to.. ignore it 😄